lstchain/reco/dl1_to_dl2.py

```diff
@@ -478,10 +478,13 @@ def build_models(filegammas, fileprotons,
         # Apply the temporary disp norm regressor and sign classifier to the test set
         disp_norm = tmp_reg_disp_norm.predict(test[config['disp_regression_features']])
         disp_sign = tmp_cls_disp_sign.predict(test[config['disp_classification_features']])
+        disp_sign_proba = tmp_cls_disp_sign.predict_proba(test[config['disp_classification_features']])
```

Comment:
There is no need to call predict and predict proba, that's wasteful. Since we now want the proba, you should change the `disp_sign` calculation to use the proba to not evaluate the random forest twice:


```
column = list(tmp_cls_disp_sign.classes_).index(1)
disp_sign = np.where(disp_sign_proba[:, col] > 0.5, 1, -1)
```

Reply:
Thanks for the comment! Actually, I just copy & paste the codes of gammaness computation, so this part can also be updated to reduce computation time
https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/reco/dl1_to_dl2.py#L652-L653